src/components/generics/vacation-test-forms/myVacationRequests.tsx

```diff
@@ -96,29 +92,72 @@ const renderVacationRequests = () => {
       maxRows={5}
       label={ strings.editorContent.leaveAComment }
       variant="outlined"
-      value={ newTextContent }
+      value={ textContent }
       onChange={ handleVacationCommentContent }
     />
   );
 
   /**
-  * Handle vacation apply button
-  */
-  const handleVacationApplyButton = () => {
-    // TODO: functionality to this
+   * Renders and calculates updated days spent for vacation
+   */
+  const renderUpdatedVacationDaysSpent = () => {
+    // Define the date range to compare with holidays
+    const holidaysFi = new Holidays("FI");
+    const startDate = new Date(selectedVacationStartDate);
+    const endDate = new Date(selectedVacationEndDate);
+    let days = 0;
+
+    // Iterate over each date in the date range and check if it is a holiday
+    for (let d = new Date(startDate); d <= endDate; d.setDate(d.getDate() + 1)) {
+      if (!holidaysFi.isHoliday(d) && d.getDay() !== 0) {
+        // eslint-disable-next-line no-plusplus
+        days++;
+      }
+    }
+    
     return (
-      console.log(`this is START DATE ${selectedVacationStartDate} and this is END DATE${selectedVacationEndDate} and this is TEXT CONTENT ${textContent}. Vacation type ${vacationType}`)
+      days
     );
   };
 
+  /**
+   * @param request 
+  * Handle vacation apply button
+  * Updates the vacation request
+  */
+  const updateRequest = async (request: any) => {
+    const changedRequest: VacationRequest = {
+      ...request,
+      startDate: selectedVacationStartDate,
+      endDate: selectedVacationEndDate,
+      type: vacationType as VacationType,
+      message: textContent,
+      updatedAt: new Date(),
+      days: renderUpdatedVacationDaysSpent()
+    };
+    if (!person) {
```

Comment:
As mentioned for the create vacation function in previous file, this could be a single line